src/elements.cpp

```diff
@@ -127,6 +129,7 @@ void G1Element::CheckValid() const {
     } catch (...) {
         throw std::invalid_argument("Relic reports invalid argument given");
     }
+    bn_free(order);
```

Comment:
technically this will leak in any of the error cases. I think this is probably good enough for a short-term fix. long term I think it would be a lot more reliable to wrap the `g1_t` and `bn_t` types in types that manage the resources.